src/k8s-extension/azext_k8s_extension/partner_extensions/AzureDefender.py

```diff
@@ -58,3 +61,23 @@ def Create(self, cmd, client, resource_group_name, cluster_name, name, cluster_t
             configuration_protected_settings=configuration_protected_settings
         )
         return extension_instance, name, create_identity
+
+    def _choose_the_right_namespace(self, cmd, cluster_resource_group_name, cluster_name, extension_name):
+        logger.warning("Choosing the right namespace ...")
+
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+        cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Kubernetes' \
+            '/connectedClusters/{2}/providers/Microsoft.KubernetesConfiguration/extensions/microsoft.azuredefender.kubernetes'.format(subscription_id, cluster_resource_group_name, cluster_name)

```

Comment:
can you pls make it generic by using the clustertype and cluster_rp for cases like provisionedClusters

Reply:
this is only applicable for connected clusters since provisioned clusters will always have the same namespace